[distgen] Allow override of global JSON via ENV (#19)
chef/go-libs: distgen/README.md

package dist
//go:generate go run github.com/chef/go-libs/distgen global.go dist https://example.com/path/to/glob_dist.json
```

> ### Using an Environment Variable

Reviewer: Documentation YESS!! Thank you for that, do you think you could update the [doc.go](https://github.com/chef/go-libs/blob/master/distgen/doc.go) as well so that we have this documentation matching our [godoc.org](https://godoc.org/github.com/chef/go-libs/distgen)

Author: Of course.  I will look at performing the addition tonight.

The usage of an environmental variable allows for better flexibility when dealing with CI systems.

> The usage inside of your dist files will not change, but you can pass ENV as of your `go generate` command.

Reviewer 2: Thanks for including documentation in your update! Just one suggestion to change as something seemed to be missing.
```suggestion
The usage inside of your dist files will not change, but you can pass ENV as a part of your `go generate` command.
```

> 

Reviewer 2: Would you please add some words explaining the example below? For example, it would be helpful to know what DIST_FILE is supposed to be, which isn't clear from the words before. Thank you in advance! :)

Author: Ok.  I would argue that it would be better to explain in more depth.  It would be more beneficial long-term to explain how CLI arguments work in the beginning of the README (only code comments at present) and how `DIST_FILE` extends them in the "Using an Environment Variable" section.

thoughts?

Reviewer: I agree that we need to document this in the README and not only in code comments. Use this section to do such documentation instead of the beginning of the README, the reason why is because this is an advance usage.

package dist
//go:generate go run github.com/chef/go-libs/distgen global.go dist https://example.com/path/to/glob_dist.json
```

### Using an Environment Variable

Using an environmental variable gives you more flexibility with CI systems.  
As we saw in the custom JSON file example, passing a URI to `go run` in the third argument 
overrides the default JSON file for `distgen`.
The `DIST_FILE` environment performs the same function by overloading the default JSON file.

> The usage inside of your dist files will not change, but you can pass ENV as a part of your `go generate` command.

Reviewer 3: ```suggestion
Using an environmental variable gives you more flexibility with CI systems. As we saw in
the custom JSON file example, passing a URI to `go run` in the third argument overrides
the default JSON file for `distgen`. The `DIST_FILE` environment performs the same
function by overloading the default JSON file. The usage inside of your dist files will
not change, but you can pass ENV as a part of your `go generate` command.
```

package dist
//go:generate go run github.com/chef/go-libs/distgen global.go dist https://example.com/path/to/glob_dist.json
```

### Using an Environment Variable

> The usage of an environmental variable allows for better flexibility when dealing with CI systems.  

Reviewer 4: ```suggestion
Using an environmental variable gives you more flexibility with CI systems.  
```

> The usage of an environmental variable allows for better flexibility when dealing with

Reviewer 4: Duplicate line

> CI systems. As we saw in the previous example, the 3rd argument that is passed to

Reviewer 4: ```suggestion
As we saw in the custom JSON file example, passing a URI to `go run` in the third argument 
```

> `go run` is a URI, which is used to override the default JSON file for `distgen`. The

Reviewer 4: ```suggestion
overrides the default JSON file for `distgen`.
```

> `DIST_FILE` environment performs the same functionality by overloading the default

Reviewer 4: ```suggestion
The `DIST_FILE` environment performs the same function by overloading the default JSON file.
```

> JSON file. The usage inside of your dist files will not change, but you can pass ENV

Reviewer 4: ```suggestion
The usage inside of your dist files will not change, but you can pass ENV as a part of your `go generate` command.
```

Can you clarify  _what_ is used in the dist files?  Do you mean the URI? Something else?

> as a part of your `go generate` command.

Reviewer 5: ```suggestion
An environmental variable will give you greater flexibility in CI systems. 
In the previous example, the 3rd argument that is passed in to `go run` is a URI which is used to override the default JSON file for `distgen`. 
The `DIST_FILE` environment performs the same function by overloading the default JSON file. 
The format of your dist files will not change, but you can include the ENV in your `go generate` command.
```
